### PR TITLE
TF-2678 implement server settings data source

### PR DESCRIPTION
**Pull request by tddang-linagora**
### Issue
- #2615 
- #2678 

### What's in this PR

- Server settings API
- Remote implementation of Server settings data source

### CODE REVIEW

File: test/features/server_settings/data/datasource_impl/remote_server_settings_data_source_impl_test.dart [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        expect(result, serverSettings);
+      });
+
+      test('should rethrow exception when ServerSettingsAPI throws exception',() async {
+        // arrange
+        when(serverSettingsAPI.updateServerSettings(any, any))
+          .thenThrow(NotFoundServerSettingsException());
+
+        // act
+
```
Comment (dab246) [inline test/features/server_settings/data/datasource_impl/remote_server_settings_data_source_impl_test.dart:77]: remove it

File: test/features/server_settings/data/datasource_impl/remote_server_settings_data_source_impl_test.dart [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        expect(result, serverSettings);
+      });
+
+      test('should rethrow exception when ServerSettingsAPI throws exception',() async {
+        // arrange
+        when(serverSettingsAPI.getServerSettings(any))
+          .thenThrow(NotFoundServerSettingsException());
+
+        // act
+
```
Comment (dab246) [inline test/features/server_settings/data/datasource_impl/remote_server_settings_data_source_impl_test.dart:49]: remove it

File: test/features/server_settings/data/datasource_impl/remote_server_settings_data_source_impl_test.dart (hunk 1 above)
Comment (tddang-linagora) [reply test/features/server_settings/data/datasource_impl/remote_server_settings_data_source_impl_test.dart]: done

File: test/features/server_settings/data/datasource_impl/remote_server_settings_data_source_impl_test.dart (hunk 2 above)
Comment (tddang-linagora) [reply test/features/server_settings/data/datasource_impl/remote_server_settings_data_source_impl_test.dart]: done

**Review by dab246 — APPROVED** [review]

File: lib/features/server_settings/data/network/server_settings_api.dart [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      .build()
+      .execute();
+
+    final updatedServerSettings = result.parse<GetServerSettingsResponse>(
+      getServerSettingsInvocation.methodCallId,
+      GetServerSettingsResponse.deserialize
+    )?.list.firstOrNull;
+
+    if (updatedServerSettings == null) {
+      throw NotFoundServerSettingsException();
```
Comment (hoangdat) [inline lib/features/server_settings/data/network/server_settings_api.dart:76]: IMO, we should throw `CanNotUpdateServerSettingsException`.

File: lib/features/server_settings/data/network/server_settings_api.dart [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    final getServerSettingsInvocation = jmapRequestBuilder.invocation(
+      getServerSettingsMethod);
+
+    final capabilities = setServerSettingsMethod.requiredCapabilities;
+
+    final result = await (jmapRequestBuilder..usings(capabilities))
+      .build()
+      .execute();
+
+    final updatedServerSettings = result.parse<GetServerSettingsResponse>(
```
Comment (hoangdat) [inline lib/features/server_settings/data/network/server_settings_api.dart:70]: what happen if `set method` have error? we should take care this case also

File: lib/features/server_settings/data/network/server_settings_api.dart (hunk 3 above)
Comment (tddang-linagora) [reply lib/features/server_settings/data/network/server_settings_api.dart]: Done

File: lib/features/server_settings/data/network/server_settings_api.dart (hunk 4 above)
Comment (tddang-linagora) [reply lib/features/server_settings/data/network/server_settings_api.dart:81]: Handled. Please review

**Review by hoangdat — APPROVED** [review]
LGTM, please rebase

**Pull request merged**